### PR TITLE
Bump sigs.k8s.io/controller-runtime to v0.17.2

### PR DESCRIPTION
**Pull request by trasc**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
- Bump sigs.k8s.io/controller-runtime to v0.17.2
- Add a SSA to fake StrategicMerge fake client patch inteceptor.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/scheduler_test.go [hunk 1]
```diff
@@ -2077,7 +2078,10 @@ func TestRequeueAndUpdate(t *testing.T) {
 			scheme := runtime.NewScheme()
 
 			objs := []client.Object{w1, q1, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "ns1"}}}
-			clientBuilder := utiltesting.NewClientBuilder().WithObjects(objs...).WithStatusSubresource(objs...)
+			clientBuilder := utiltesting.NewClientBuilder().
+				WithInterceptorFuncs(interceptor.Funcs{SubResourcePatch: utiltesting.TreatSSAAsStrategicMerge}).
+				WithObjects(objs...).
+				WithStatusSubresource(objs...)
 			cl := clientBuilder.Build()
```
[the remark this answers is not in the document]
Comment (trasc) [reply pkg/scheduler/scheduler_test.go]: done

**Review by tenzen-y — COMMENTED** [review]
/approve

**Comment by alculquicondor** [comment]
/cherry-pick release-0.6

For #2271

**Comment by alculquicondor** [comment]
@trasc can you prepare a cherry-pick of this?

**Comment by marcellio835** [comment]
Legal

**Pull request merged**